refactor(parser): Improve SQS models with examples and descriptions

dcabib opened the pull request:
closes #7116

## Summary

Enhances the SQS parser models with field descriptions and examples using Pydantic's Field() functionality. This improvement provides better documentation and metadata for SQS event parsing, following the pattern established in PR #7100.

### Changes

- Added Field() descriptions and examples to all SQS parser model classes:
  - `SqsAttributesModel` - Core SQS message attributes with AWS-accurate descriptions
  - `SqsMsgAttributeModel` - User-defined message attributes with comprehensive documentation  
  - `SqsRecordModel` - Individual SQS record structure with detailed field explanations
  - `SqsModel` - Root SQS event model with proper sequence documentation

- All field descriptions are based on official AWS SQS documentation
- Examples sourced from actual test events (sqsEvent.json, sqsDlqTriggerEvent.json)
- Maintains 100% backward compatibility - no breaking changes to type annotations or validation logic

### User experience

Customers will be able to see examples and descriptions when using SQS parser models, providing:
- Rich IntelliSense with field descriptions and realistic examples
- Self-documenting code without needing external AWS documentation  
- Professional API documentation via .model_json_schema()
- Faster development with immediate reference for acceptable values

## Checklist

- [x] Meet tenets criteria
- [x] I have performed a self-review of this change
- [x] Changes have been tested (all existing tests pass)
- [x] Changes are documented (via Field descriptions)
- [x] PR title follows conventional commit semantics

**Is this a breaking change?** No

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

Closes #7116

dcabib committed:
refactor(parser): Improve SQS models with examples and descriptions

Enhances the SQS parser models with field descriptions and examples using Pydantic's Field() functionality. This improvement provides better documentation and metadata for SQS event parsing, following the pattern established in PR #7100.

All field descriptions are based on official AWS SQS documentation and include realistic examples from actual test events.

Closes #7116

leandrodamascena commented on aws_lambda_powertools/utilities/parser/models/sqs.py:
```diff
@@ -49,17 +100,77 @@ class SqsMsgAttributeModel(BaseModel):
+        examples=[
+            "AQEBwJnKyrHigUMZj6rYigCgxlaS3SLy0a...",
+            "AQEBzWwaftRI0KuVm4tP+/7q1rGgNqicHq...",
+            "AQEBl1pqxv+ZHkarVAWZUyWgj2mmqJGLBTo6YFOi/bw1QpBTpJBGJPLOTZrjKztKIbAB8EXkG7zHlbkn+Ze/AHMKKuhST9azHu8LyF4Ffu9uPkZc5xzggXlfFBWH3TUKyV+F5Obaj3esyX8YfM/zfgjbRuu5nc2tfPhvaSYEaTZsdMpzIB5tyKvHxAltLxK7upRHeoT768M9UrFYswarFTBn8piDbnsPsUhi8Q9G4Q4xSI0fLQANmryBsRJIzGQTVxenDad+MJ7XEL+hD3p2DmW+ycvv6WD7bdedqQuroQG8+ca1Dz7s3CBbXw9ZZnUziPa7LH1j1Lky5bAxpNF+BlurRS9pFBnomhwpylrGxtGfaEmUW1G7jnrG97sZNOLOFU ...
```
Hey @dcabib pls remove this line.

leandrodamascena reviewed (commented):
Hey @dcabib thanks a lot! Just a small change and we are ready to merge.

leandrodamascena commented:
Hey @dcabib please run `make format` to format the file and fix the CI.

dcabib committed:
style: remove long receiptHandle example and apply formatting

Remove extremely long receiptHandle example as requested by leandrodamascena and apply ruff formatting to fix CI issues.

leandrodamascena committed:
Fix styles

leandrodamascena committed:
Merge branch 'develop' into feat/parser-improve-sqs-models

leandrodamascena reviewed (approved):
Thanks @dcabib! Approved!
